The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6882081015/job/18719819928) [(plain)](https://github.com/rust-lang/rust/commit/984898da1727158f096c3a07ebd9a5ac487d733f/checks/18719819928/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=aeubanks
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_46513188-9898-4b3d-8d8b-3bd1f5728c06
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=6646347a6cf1d57425ef04596edfddef94511150
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_46513188-9898-4b3d-8d8b-3bd1f5728c06
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_46513188-9898-4b3d-8d8b-3bd1f5728c06
GITHUB_TRIGGERING_ACTOR=aeubanks
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/117948/merge
GITHUB_WORKFLOW_SHA=6646347a6cf1d57425ef04596edfddef94511150
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
   Compiling cstr v0.2.8
   Compiling time-core v0.1.1
The following warnings were emitted during compilation:

warning: llvm-wrapper/RustWrapper.cpp: In function ‘LLVMOpaqueMetadata* LLVMRustDIBuilderCreateStaticMemberType(LLVMRustDIBuilderRef, LLVMMetadataRef, const char*, size_t, LLVMMetadataRef, unsigned int, LLVMMetadataRef, LLVMRustDIFlags, LLVMValueRef, uint32_t)’:
warning: llvm-wrapper/RustWrapper.cpp:978:46: error: no matching function for call to ‘llvm::DIBuilder::createStaticMemberType(llvm::DIScope*, llvm::StringRef, llvm::DIFile*, unsigned int&, llvm::DIType*, llvm::DINode::DIFlags, llvm::ConstantInt*, llvm::dwarf::Tag, uint32_t&)’
warning:   978 |   return wrap(Builder->createStaticMemberType(
warning:       |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
warning:   979 |     unwrapDI<DIDescriptor>(Scope),
warning:       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~            
warning:   980 |     StringRef(Name, NameLen),
warning:       |     ~~~~~~~~~~~~~~~~~~~~~~~~~                 
warning:   981 |     unwrapDI<DIFile>(File),
warning:   982 |     LineNo,
warning:       |     ~~~~~~~                                   
warning:       |     ~~~~~~~                                   
warning:   983 |     unwrapDI<DIType>(Ty),
warning:       |     ~~~~~~~~~~~~~~~~~~~~~                     
warning:   984 |     fromRust(Flags),
warning:       |     ~~~~~~~~~~~~~~~~                          
warning:   985 |     unwrap<llvm::ConstantInt>(val),
warning:       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~           
warning:   986 | #if LLVM_VERSION_GE(17, 0)
warning:       | ~~~~~~~~~~~~~~~~~~~~~~~~~~                    
warning:   987 |     llvm::dwarf::DW_TAG_member,
warning:   988 | #endif
warning:       | ~~~~~~                                        
warning:   989 |     AlignInBits
warning:       |     ~~~~~~~~~~~                               
warning:       |     ~~~~~~~~~~~                               
warning:   990 |   ));
warning:       |   ~                                           
warning: In file included from llvm-wrapper/LLVMWrapper.h:40,
warning:                  from llvm-wrapper/RustWrapper.cpp:1:
warning: /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/include/llvm/IR/DIBuilder.h:380:20: note: candidate: ‘llvm::DIDerivedType* llvm::DIBuilder::createStaticMemberType(llvm::DIScope*, llvm::StringRef, llvm::DIFile*, unsigned int, llvm::DIType*, llvm::DINode::DIFlags, llvm::Constant*, uint32_t)’
warning:   380 |     DIDerivedType *createStaticMemberType(DIScope *Scope, StringRef Name,
warning: /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/include/llvm/IR/DIBuilder.h:380:20: note:   candidate expects 8 arguments, 9 provided

error: failed to run custom build command for `rustc_llvm v0.0.0 (/checkout/compiler/rustc_llvm)`


Caused by:
  process didn't exit successfully: `/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/release/build/rustc_llvm-8cafddd5c0ba7bb5/build-script-build` (exit status: 1)
  --- stdout
  cargo:rustc-check-cfg=values(llvm_component,"ipo")
  cargo:rustc-check-cfg=values(llvm_component,"bitreader")
  cargo:rustc-check-cfg=values(llvm_component,"bitwriter")
  cargo:rustc-check-cfg=values(llvm_component,"linker")
  cargo:rustc-check-cfg=values(llvm_component,"asmparser")
  cargo:rustc-check-cfg=values(llvm_component,"lto")
  cargo:rustc-check-cfg=values(llvm_component,"coverage")
  cargo:rustc-check-cfg=values(llvm_component,"instrumentation")
  cargo:rustc-check-cfg=values(llvm_component,"x86")
  cargo:rustc-check-cfg=values(llvm_component,"arm")
  cargo:rustc-check-cfg=values(llvm_component,"aarch64")
  cargo:rustc-check-cfg=values(llvm_component,"amdgpu")
  cargo:rustc-check-cfg=values(llvm_component,"avr")
  cargo:rustc-check-cfg=values(llvm_component,"loongarch")
  cargo:rustc-check-cfg=values(llvm_component,"m68k")
  cargo:rustc-check-cfg=values(llvm_component,"csky")
  cargo:rustc-check-cfg=values(llvm_component,"mips")
  cargo:rustc-check-cfg=values(llvm_component,"powerpc")
  cargo:rustc-check-cfg=values(llvm_component,"systemz")
  cargo:rustc-check-cfg=values(llvm_component,"jsbackend")
  cargo:rustc-check-cfg=values(llvm_component,"webassembly")
  cargo:rustc-check-cfg=values(llvm_component,"msp430")
  cargo:rustc-check-cfg=values(llvm_component,"sparc")
  cargo:rustc-check-cfg=values(llvm_component,"nvptx")
  cargo:rustc-check-cfg=values(llvm_component,"hexagon")
  cargo:rustc-check-cfg=values(llvm_component,"riscv")
  cargo:rustc-check-cfg=values(llvm_component,"bpf")
  cargo:rerun-if-env-changed=RUST_CHECK
  cargo:rerun-if-env-changed=REAL_LIBRARY_PATH_VAR
  cargo:rerun-if-env-changed=REAL_LIBRARY_PATH
  cargo:rerun-if-env-changed=LLVM_CONFIG
  cargo:rerun-if-changed=/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/bin/llvm-config
  cargo:rustc-cfg=llvm_component="aarch64"
  cargo:rustc-cfg=llvm_component="arm"
  cargo:rustc-cfg=llvm_component="asmparser"
  cargo:rustc-cfg=llvm_component="avr"
  cargo:rustc-cfg=llvm_component="bitreader"
  cargo:rustc-cfg=llvm_component="bitwriter"
  cargo:rustc-cfg=llvm_component="bpf"
  cargo:rustc-cfg=llvm_component="coverage"
  cargo:rustc-cfg=llvm_component="csky"
  cargo:rustc-cfg=llvm_component="hexagon"
  cargo:rustc-cfg=llvm_component="instrumentation"
  cargo:rustc-cfg=llvm_component="ipo"
  cargo:rustc-cfg=llvm_component="linker"
  cargo:rustc-cfg=llvm_component="loongarch"
  cargo:rustc-cfg=llvm_component="lto"
  cargo:rustc-cfg=llvm_component="m68k"
  cargo:rustc-cfg=llvm_component="mips"
  cargo:rustc-cfg=llvm_component="msp430"
  cargo:rustc-cfg=llvm_component="nvptx"
  cargo:rustc-cfg=llvm_component="powerpc"
  cargo:rustc-cfg=llvm_component="riscv"
  cargo:rustc-cfg=llvm_component="sparc"
  cargo:rustc-cfg=llvm_component="systemz"
  cargo:rustc-cfg=llvm_component="webassembly"
  cargo:rustc-cfg=llvm_component="x86"
  cargo:rerun-if-env-changed=LLVM_RUSTLLVM
  cargo:rerun-if-env-changed=LLVM_NDEBUG
  cargo:rerun-if-changed=llvm-wrapper/README
  cargo:rerun-if-changed=llvm-wrapper/SymbolWrapper.cpp
  cargo:rerun-if-changed=llvm-wrapper/RustWrapper.cpp
  cargo:rerun-if-changed=llvm-wrapper/Linker.cpp
  cargo:rerun-if-changed=llvm-wrapper/.editorconfig
  cargo:rerun-if-changed=llvm-wrapper/CoverageMappingWrapper.cpp
  cargo:rerun-if-changed=llvm-wrapper/PassWrapper.cpp
  cargo:rerun-if-changed=llvm-wrapper/ArchiveWrapper.cpp
  cargo:rerun-if-changed=llvm-wrapper/LLVMWrapper.h
  TARGET = Some("x86_64-unknown-linux-gnu")
  HOST = Some("x86_64-unknown-linux-gnu")
  cargo:rerun-if-env-changed=CXX_x86_64-unknown-linux-gnu
  CXX_x86_64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CXX_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=CXX_x86_64_unknown_linux_gnu
  CXX_x86_64_unknown_linux_gnu = Some("sccache c++")
  CXXFLAGS_x86_64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CXXFLAGS_x86_64_unknown_linux_gnu
  CXXFLAGS_x86_64_unknown_linux_gnu = Some("-ffunction-sections -fdata-sections -fPIC -m64")
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running: "sccache" "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/include" "-std=c++17" "-fno-exceptions" "-funwind-tables" "-fno-rtti" "-D_GNU_SOURCE" "-D_DEBUG" "-D_GLIBCXX_ASSERTIONS" "-D_LIBCPP_ENABLE_HARDENED_MODE" "-D__STDC_CONSTANT_MACROS" "-D__STDC_FORMAT_MACROS" "-D__STDC_LIMIT_MACROS" "-DLLVM_COMPONENT_AARCH64" "-DLLVM_COMPONENT_ARM" "-DLLVM_COMPONENT_ASMPARSER" "-DLLVM_COMPONENT_AVR" "-DLLVM_COMPONENT_BITREADER" "-DLLVM_COMPONENT_BITWRITER" "-DLLVM_COMPONENT_BPF" "-DLLVM_COMPONENT_COVERAGE" "-DLLVM_COMPONENT_CSKY" "-DLLVM_COMPONENT_HEXAGON" "-DLLVM_COMPONENT_INSTRUMENTATION" "-DLLVM_COMPONENT_IPO" "-DLLVM_COMPONENT_LINKER" "-DLLVM_COMPONENT_LOONGARCH" "-DLLVM_COMPONENT_LTO" "-DLLVM_COMPONENT_M68K" "-DLLVM_COMPONENT_MIPS" "-DLLVM_COMPONENT_MSP430" "-DLLVM_COMPONENT_NVPTX" "-DLLVM_COMPONENT_POWERPC" "-DLLVM_COMPONENT_RISCV" "-DLLVM_COMPONENT_SPARC" "-DLLVM_COMPONENT_SYSTEMZ" "-DLLVM_COMPONENT_WEBASSEMBLY" "-DLLVM_COMPONENT_X86" "-DLLVM_RUSTLLVM" "-DNDEBUG" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/build/rustc_llvm-2cdecfa9d2205cfd/out/llvm-wrapper/PassWrapper.o" "-c" "llvm-wrapper/PassWrapper.cpp"
  exit status: 0
  running: "sccache" "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/include" "-std=c++17" "-fno-exceptions" "-funwind-tables" "-fno-rtti" "-D_GNU_SOURCE" "-D_DEBUG" "-D_GLIBCXX_ASSERTIONS" "-D_LIBCPP_ENABLE_HARDENED_MODE" "-D__STDC_CONSTANT_MACROS" "-D__STDC_FORMAT_MACROS" "-D__STDC_LIMIT_MACROS" "-DLLVM_COMPONENT_AARCH64" "-DLLVM_COMPONENT_ARM" "-DLLVM_COMPONENT_ASMPARSER" "-DLLVM_COMPONENT_AVR" "-DLLVM_COMPONENT_BITREADER" "-DLLVM_COMPONENT_BITWRITER" "-DLLVM_COMPONENT_BPF" "-DLLVM_COMPONENT_COVERAGE" "-DLLVM_COMPONENT_CSKY" "-DLLVM_COMPONENT_HEXAGON" "-DLLVM_COMPONENT_INSTRUMENTATION" "-DLLVM_COMPONENT_IPO" "-DLLVM_COMPONENT_LINKER" "-DLLVM_COMPONENT_LOONGARCH" "-DLLVM_COMPONENT_LTO" "-DLLVM_COMPONENT_M68K" "-DLLVM_COMPONENT_MIPS" "-DLLVM_COMPONENT_MSP430" "-DLLVM_COMPONENT_NVPTX" "-DLLVM_COMPONENT_POWERPC" "-DLLVM_COMPONENT_RISCV" "-DLLVM_COMPONENT_SPARC" "-DLLVM_COMPONENT_SYSTEMZ" "-DLLVM_COMPONENT_WEBASSEMBLY" "-DLLVM_COMPONENT_X86" "-DLLVM_RUSTLLVM" "-DNDEBUG" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/build/rustc_llvm-2cdecfa9d2205cfd/out/llvm-wrapper/RustWrapper.o" "-c" "llvm-wrapper/RustWrapper.cpp"
  cargo:warning=llvm-wrapper/RustWrapper.cpp: In function ‘LLVMOpaqueMetadata* LLVMRustDIBuilderCreateStaticMemberType(LLVMRustDIBuilderRef, LLVMMetadataRef, const char*, size_t, LLVMMetadataRef, unsigned int, LLVMMetadataRef, LLVMRustDIFlags, LLVMValueRef, uint32_t)’:
  cargo:warning=llvm-wrapper/RustWrapper.cpp:978:46: error: no matching function for call to ‘llvm::DIBuilder::createStaticMemberType(llvm::DIScope*, llvm::StringRef, llvm::DIFile*, unsigned int&, llvm::DIType*, llvm::DINode::DIFlags, llvm::ConstantInt*, llvm::dwarf::Tag, uint32_t&)’
  cargo:warning=  978 |   return wrap(Builder->createStaticMemberType(
  cargo:warning=      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  cargo:warning=  979 |     unwrapDI<DIDescriptor>(Scope),
  cargo:warning=      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~            
  cargo:warning=  980 |     StringRef(Name, NameLen),
  cargo:warning=      |     ~~~~~~~~~~~~~~~~~~~~~~~~~                 
  cargo:warning=  981 |     unwrapDI<DIFile>(File),
  cargo:warning=      |     ~~~~~~~~~~~~~~~~~~~~~~~                   
  cargo:warning=  982 |     LineNo,
  cargo:warning=      |     ~~~~~~~                                   
  cargo:warning=  983 |     unwrapDI<DIType>(Ty),
  cargo:warning=      |     ~~~~~~~~~~~~~~~~~~~~~                     
  cargo:warning=  984 |     fromRust(Flags),
  cargo:warning=      |     ~~~~~~~~~~~~~~~~                          
  cargo:warning=  985 |     unwrap<llvm::ConstantInt>(val),
  cargo:warning=      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~           
  cargo:warning=  986 | #if LLVM_VERSION_GE(17, 0)
  cargo:warning=      | ~~~~~~~~~~~~~~~~~~~~~~~~~~                    
  cargo:warning=  987 |     llvm::dwarf::DW_TAG_member,
  cargo:warning=      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~               
  cargo:warning=  988 | #endif
  cargo:warning=      | ~~~~~~                                        
  cargo:warning=  989 |     AlignInBits
  cargo:warning=      |     ~~~~~~~~~~~                               
  cargo:warning=  990 |   ));
  cargo:warning=      |   ~                                           
  cargo:warning=In file included from llvm-wrapper/LLVMWrapper.h:40,
  cargo:warning=                 from llvm-wrapper/RustWrapper.cpp:1:
  cargo:warning=/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/include/llvm/IR/DIBuilder.h:380:20: note: candidate: ‘llvm::DIDerivedType* llvm::DIBuilder::createStaticMemberType(llvm::DIScope*, llvm::StringRef, llvm::DIFile*, unsigned int, llvm::DIType*, llvm::DINode::DIFlags, llvm::Constant*, uint32_t)’
  cargo:warning=  380 |     DIDerivedType *createStaticMemberType(DIScope *Scope, StringRef Name,
  cargo:warning=      |                    ^~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/include/llvm/IR/DIBuilder.h:380:20: note:   candidate expects 8 arguments, 9 provided

  --- stderr



  error occurred: Command "sccache" "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/include" "-std=c++17" "-fno-exceptions" "-funwind-tables" "-fno-rtti" "-D_GNU_SOURCE" "-D_DEBUG" "-D_GLIBCXX_ASSERTIONS" "-D_LIBCPP_ENABLE_HARDENED_MODE" "-D__STDC_CONSTANT_MACROS" "-D__STDC_FORMAT_MACROS" "-D__STDC_LIMIT_MACROS" "-DLLVM_COMPONENT_AARCH64" "-DLLVM_COMPONENT_ARM" "-DLLVM_COMPONENT_ASMPARSER" "-DLLVM_COMPONENT_AVR" "-DLLVM_COMPONENT_BITREADER" "-DLLVM_COMPONENT_BITWRITER" "-DLLVM_COMPONENT_BPF" "-DLLVM_COMPONENT_COVERAGE" "-DLLVM_COMPONENT_CSKY" "-DLLVM_COMPONENT_HEXAGON" "-DLLVM_COMPONENT_INSTRUMENTATION" "-DLLVM_COMPONENT_IPO" "-DLLVM_COMPONENT_LINKER" "-DLLVM_COMPONENT_LOONGARCH" "-DLLVM_COMPONENT_LTO" "-DLLVM_COMPONENT_M68K" "-DLLVM_COMPONENT_MIPS" "-DLLVM_COMPONENT_MSP430" "-DLLVM_COMPONENT_NVPTX" "-DLLVM_COMPONENT_POWERPC" "-DLLVM_COMPONENT_RISCV" "-DLLVM_COMPONENT_SPARC" "-DLLVM_COMPONENT_SYSTEMZ" "-DLLVM_COMPONENT_WEBASSEMBLY" "-DLLVM_COMPONENT_X86" "-DLLVM_RUSTLLVM" "-DNDEBUG" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/build/rustc_llvm-2cdecfa9d2205cfd/out/llvm-wrapper/RustWrapper.o" "-c" "llvm-wrapper/RustWrapper.cpp" with args "c++" did not execute successfully (status code exit status: 1).

warning: build failed, waiting for other jobs to finish...
Build completed unsuccessfully in 0:01:04
cat: /tmp/toolstate/toolstates.json: No such file or directory
```

</details>

--- a/compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp
+++ b/compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp
@@ -983,7 +983,7 @@ extern "C" LLVMMetadataRef LLVMRustDIBuilderCreateStaticMemberType(
     unwrapDI<DIType>(Ty),
     fromRust(Flags),
     unwrap<llvm::ConstantInt>(val),
-#if LLVM_VERSION_GE(17, 0)
+#if LLVM_VERSION_GE(18, 0)
     llvm::dwarf::DW_TAG_member,
 #endif
     AlignInBits

